esp-hal/src/uart/mod.rs

```diff
@@ -863,6 +860,43 @@ where
         while !self.is_tx_idle() {}
     }
 
+    /// Sends a break signal for a specified duration in bit time.
+    ///
+    /// Duration is in bits, the time it takes to transfer one bit at the
+    /// current baud rate. The delay during the break is just is busy-waiting.
+    #[instability::unstable]
+    pub fn send_break(&mut self, bits: u32) {
+        // Read the current TX inversion state
+        let original_txd_inv = self.uart.info().regs().conf0().read().txd_inv().bit();
+
+        // Invert the TX line (toggle the current state)
+        self.uart
+            .info()
+            .regs()
+            .conf0()
+            .modify(|_, w| w.txd_inv().bit(!original_txd_inv));
+
+        #[cfg(any(esp32c6, esp32h2))]
```

Comment:
According to the TRMs, both C3 and S3 reqire a sync after writing CONF0.

Reply:
Resolved in [07e3e13](https://github.com/esp-rs/esp-hal/pull/4284/commits/07e3e13401952f1776a4c02d5315c465f3dc6a3f)